bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import datetime
+import logging
+from typing import List, Optional
+
+from discord import Color, Embed, Message
+from discord.ext import commands
+
+from bot.constants import Wikipedia
+
+log = logging.getLogger(__name__)
+
+SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch={search_term}&format=json"
+WIKIPEDIA_URL = "https://en.wikipedia.org/wiki/{title}"
+
+
+class WikipediaCog(commands.Cog):
+    """Get info from wikipedia."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_wikipedia(self, search_term: str) -> Optional[List[str]]:
+        """Search wikipedia and return the first page found."""
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+        if len(search_results) == 0:
+            return None
+
+        # we dont like "may refere to" pages.
+        for search_result in search_results:
+            log.info("trying to append titles")
+            if "may refer to" in search_result["snippet"]:
+                pass
+            else:
+                page.append(search_result["title"])
+        log.info("Finished appending titles")
+        return page
+
+    @commands.cooldown(1, 10, commands.BucketType.user)
+    @commands.command(name="wikipedia", aliases=["wiki"])
+    async def wikipedia_search_command(self, ctx: commands.Context, *, search: str) -> None:
+        """Returns list of your search query from wikipedia."""
+        titles_no_underscore: List[str] = []
+        s_desc = ''
+
+        titles = await self.search_wikipedia(search)
+
+        def check(message: Message) -> bool:
+            return message.author.id == ctx.author.id
+
+        if titles is None:
```

Comment:
```suggestion
        if not titles:
```
Checks for empty strings, empty lists etc as well.